Code/Tools/ProjectManager/Source/PythonBindings.cpp

```diff
@@ -675,6 +675,14 @@ namespace O3DE::ProjectManager
                     }
                 }
 
+                if (data.contains("dependencies"))
+                {
+                    for (auto dependency : data["dependencies"])
+                    {
+                        gemInfo.m_dependencies.push_back(Py_To_String(dependency));
+                    }
+                }
+
```

Comment:
There is still a gap of not being able to recursively grab the gem dependencies.

How about adding a function to the o3de python package manifest.py script to actually recurse through gem dependencies?

The `manifest.get_registered` function can be invoked with the dependent gem_name and the project_path to gather the path to the gem itself.
In which case the recursion can occur.

Reply:
Currently, the gem catalog resolves the full recursive gem dependencies in GemModel::UpdateGemDependencies() using the recursive GetAllDependingGems() function, which is also why the unit test works.

I like the idea of adding that functionality to the o3de python package too though.  Will look into that too.

Reply:
I'm having second thoughts about moving the recursive dependency gather function to python in this review, which has gotten much larger than I intended, lets implement it when we have a use for it in the CLI.

Reply:
I actually disagree with that.
Because the functionality to determine dependencies shouldn't be in the UI, if it is not in the CLI.

Having the UI have it's own function to gathering dependencies provides a mechanism where if it is added later, code would be duplicated between the ProjectManager and the o3de python scripts with the o3de python scripts being the definitive place for this logic.

If the functionality isn't going to be added to the CLI at this time, I think we should add an issue to add a `get_all_dependent_gems` function to the o3de `manifest.py` script and to update the [GemModel.cpp](https://github.com/o3de/o3de/blob/ff188b74f5e73b69bd1ba00a4325ea9f3f32bad2/Code/Tools/ProjectManager/Source/GemCatalog/GemModel.cpp#L178-L190) to use it

Reply:
The functionality to determine dependencies is in the CLI, but it provides the immediate gem dependencies, the current UI is just generating a flattened map from that tree for fast lookup, recursive dependencies and determining reverse dependencies.  In other words, the data is the same, but in different structures useful for filtering, searching etc.

I need to check with UX if they want to display the full recursive dependencies in the Gem Inspector panel - I had assumed they only wanted to display what's in the gem.json, but thinking about it now, it'd be more useful to see the full recursive list of gems, so if the gem.json depends on atom you know the gem is bringing in all the gems atom depends on.

Reply:
Wouldn't it be possible to determine that for the user by looking at dependent Gems?

So if the user is looking at Atom, and Sees that it depends on EMotionFXAtom, they can look at EMotionFXAtom gem depends on EMotionFX.

Reply:
I got some time to experiment with moving the get_gem_dependencies function to Python and it's working, but my initial approach was to call get_all_dependent_gems() per gem and was very slow (10+secs).  According to cProfile most of the time was due to many calls to get_gem_json_data(gem_name/gem_path).  By using caching and retrieving all the gem dependencies in a single get_all_gem_dependencies() function I was able to get the time under a second.  Still not as fast as the current implementation though which benefits from already having all the gem.json info in memory (and it's c++). 

If I we really wanted to, we could make a combined method that retrieved all the gem.json info and full dependency lists in one python call, but the UI would still have to massage the data to convert gem name strings to QModelIndex GemModel entries.

We can consider these findings when considering the implementation and use of the python function in the future.

GitHub Issue: https://github.com/o3de/o3de/issues/4301

This was my latest iteration of the python function:
```
def get_all_gem_dependencies(project_path: pathlib.Path = None) -> set:
    gems = get_all_gems(project_path)

    # cache all the immediate dependencies to avoid searching for and parsing 
    # gem.json many times
    gem_immediate_dependencies = {}
    for gem_path in gems:
        gem_json_data = get_gem_json_data(gem_path=gem_path, project_path=project_path)
        gem_name = gem_json_data['gem_name']
        if 'dependencies' in gem_json_data:
            gem_immediate_dependencies[gem_name] = gem_json_data['dependencies']
        else:
            gem_immediate_dependencies[gem_name] = []

    gem_all_dependencies = {}
    gem_dependencies = set()

    @lru_cache(maxsize=None)
    def _get_dependencies(gem_name:str):
        for dependency in gem_immediate_dependencies[gem_name]:
            # avoid circular dependencies
            if dependency not in gem_dependencies:
                gem_dependencies.add(dependency)
                _get_dependencies(gem_name=dependency)
        return gem_dependencies

    for gem_name in gem_immediate_dependencies:
        gem_dependencies = set()
        gem_all_dependencies[gem_name] = _get_dependencies(gem_name=gem_name)

    return gem_all_dependencies
```